extensions/sql-database-projects/src/models/project.ts

```diff
@@ -866,6 +866,41 @@ export class Project implements ISqlProject {
 		await this.addToProjFile(sqlCmdVariableEntry);
 	}
 
+	/**
+	 * Creates DatabaseSource tag if one doesn't exist and adds database source
+	 * to the collection of existing sources
+	 * @param databaseSource Source of the database to add
+	 */
+	public async addDatabaseSource(databaseSource: string): Promise<void> {
+		await this.addValueToCollectionProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	 * Removes database source from the collection of existing sources and removes
+	 * DatabaseSource tag from the project if no sources remain
+	 * @param databaseSource Source of the database to remove
+	 */
+	public async removeDatabaseSource(databaseSource: string): Promise<void> {
+		await this.removeValueFromCollectionProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	 * Get array of all database source values, regardless of if value is 'well-know'.
+	 * @returns Array of all database source values, regardless of if value is 'well-know'
+	 */
+	public getDatabaseSourceValues(): string[] {
+		// TODO: Support should be added for values that contain ';'
+
+		const databaseSource = this.evaluateProjectPropertyValue(constants.DatabaseSource);
+
+		if (databaseSource !== undefined) {
+			return databaseSource.split(';')
+				.filter(value => value.length > 0);
+		}
+
+		return [];
```

Comment:
Cleaner to keep the primary path at the top level and dump out in the exceptional case:

```
if (databaseSource === undefined) {
   return [];
}

...
```